src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -1703,4 +1703,47 @@ public static Boolean areItemStacksMergable(ItemStack existingStack, ItemStack m
 
         return existingStack.getMaxStackSize() >= (getItemStackSize(existingStack) + getItemStackSize(mergingStack));
     }
+
+    /**
+     * Remove a list of stacks from a given Itemhandler
+     * @param handler the itemHandler.
+     * @param input the list of stacks
+     */
+    public static boolean removeStacksFromItemHandler(final IItemHandler handler, final List<ItemStack> input)
```

Comment:
Please add a method to do the same for ICapabilityProviders.
See the other methods in InventoryUtils that wrap IItemHandler from ICapabilityProviders.